api/src/main/java/org/openmrs/serialization/SimpleXStreamSerializer.java

```diff
@@ -40,39 +47,91 @@
  * </code>
  *
  */
+@Component("simpleXStreamSerializer")
 public class SimpleXStreamSerializer implements OpenmrsSerializer {
+
+	protected static final Logger log = LoggerFactory.getLogger(SimpleXStreamSerializer.class);
 	
 	// cached xstream object
-	public XStream xstream = null;
+	private volatile XStream xstream;
+	
+	private final XStream customXStream;
+	
+	private AdministrationService adminService;
 	
 	/**
 	 * Default Constructor
 	 *
 	 * @throws SerializationException
 	 */
 	public SimpleXStreamSerializer() throws SerializationException {
-		this(null);
+		this(null, null);
 	}
-	
+
 	/**
-	 * Constructor that takes a custom XStream object
-	 * @param customXstream
+	 * Constructor that takes a custom XStream object. 
+	 * 
+	 * Please note that it is deprecated since it now requires AdministrationService to fully configure
+	 * whitelists defined via GPs.
+	 * 
+	 * @deprecated since 2.7.0, 2.6.2, 2.5.13 use SimpleXStreamSerializer(XStream, AdministrationService)
+	 * @param customXStream
 	 * @throws SerializationException
 	 */
-	public SimpleXStreamSerializer(XStream customXstream) throws SerializationException {
-		if (customXstream == null) {
-			
-			xstream = new XStream();
-			
+	public SimpleXStreamSerializer(XStream customXStream) throws SerializationException {
+		this(customXStream, null);
+	}
+	
+	public SimpleXStreamSerializer(XStream customXStream, AdministrationService adminService) throws SerializationException {
+		this.customXStream = customXStream;
+		this.adminService = adminService;
+	}
+	
+	@Autowired
+	public SimpleXStreamSerializer(AdministrationService adminService) {
+		this.customXStream = null;
+		this.adminService = adminService;
+	}
+
+	/**
+	 * Setups XStream security using AdministrationService.getSerializerWhitelistTypes()
+	 * 
+	 * @since 2.7.0, 2.6.2, 2.5.13 
+	 * @param xstream
+	 * @param adminService
+	 */
+	public static void setupXStreamSecurity(XStream xstream, AdministrationService adminService) {
+		xstream.allowTypesByWildcard(new String[] {"org.openmrs.**"});
```

Comment:
Why add this here, if we are also adding this as a hard-coded value within the getSerializerWhitelistTypes method?  Can we just add the fixed, hard-coded values in one place?

Reply:
It's duplicated in cases when adminService is not passed so we still have some basic setup. I'll move that line to the else statement just to be explicit with the intention.